pkg/strategy/grid2/strategy.go

```diff
@@ -1987,16 +1987,19 @@ func (s *Strategy) Run(ctx context.Context, _ bbgo.OrderExecutor, session *bbgo.
 
 	session.UserDataStream.OnAuth(func() {
 		if !bbgo.IsBackTesting {
-			// callback may block the stream execution, so we spawn the recover function to the background
-			// add (5 seconds + random <10 seconds jitter) delay
-			go time.AfterFunc(util.MillisecondsJitter(5*time.Second, 1000*10), func() {
-				recovered := atomic.LoadInt32(&s.recovered)
-				if recovered == 0 {
-					return
-				}
-
-				s.recoverActiveOrders(ctx, session)
-			})
+			go s.recoverActiveOrdersWithOpenOrdersPeriodically(ctx)
```

Comment:
every "OnAuth" event will launch a new goroutine for this?

Reply:
you might need to check if `recovered := atomic.LoadInt32(&s.recovered)` then we start the goroutine for this?

or maybe you can move this to OnStart (which only trigger one time)

Reply:
Yes, I should move it to OnStart